timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosRemotingUtils.java

```diff
@@ -65,8 +66,8 @@ public static List<String> addProtocols(ClusterConfiguration cluster, List<Strin
 
     public static URL convertAddressToUrl(ClusterConfiguration cluster, String address) {
         try {
-            return UriBuilder.fromPath(addProtocol(cluster, address)).build().toURL();
-        } catch (MalformedURLException e) {
+            return new URIBuilder(addProtocol(cluster, address)).build().toURL();
+        } catch (MalformedURLException | URISyntaxException e) {
```

Comment:
Can we just create the URL directly since `addProtocol` handles prefixing the scheme of `https` or `http`?

```suggestion
            return new URL(addProtocol(cluster, address));
        } catch (MalformedURLException e) {
```

Reply:
You said it better :P https://github.com/palantir/atlasdb/pull/6972#discussion_r1492629376